Lib/asyncio/tasks.py

```diff
@@ -640,9 +640,34 @@ def __sleep0():
     yield
 
 
+def _check_delay(delay):
+    """Check if the value of 'delay' is valid."""
+
+    import sys
+    import math
+
+    SEC_TO_NS = 1000 * 1000 * 1000
+
+    if not isinstance(delay, (int, float)):
+        raise TypeError(f"{type(delay)} object cannot be interpreted as an integer")
+
+    # According to Modules/timemodule.c
+    if (delay > sys.maxsize / SEC_TO_NS):
```

Comment:
It is incorrect because of 32-bit platform.
`sys.maxsize`  on 32-bit platform equal to `2 ** 31 -1`.
So, in that case `sys.maxsize / SEC_TO_NS` is equal to `2.147483647`

Reply:
Oh!
I forgot about this matter.
In `timemodule.c`, INT64_MAX is used and is not subject to the limitation of 32-bit.
If I write it as `INT64_MAX = 2**64 - 1`, how would it be?

`if (delay > INT64_MAX / SEC_TO_NS)`

Reply:
> Oh! I forgot about this matter. In `timemodule.c`, INT64_MAX is used and is not subject to the limitation of 32-bit. If I write it as `INT64_MAX = 2**64 - 1`, how would it be?
> 
> `if (delay > INT64_MAX / SEC_TO_NS)`

I think it's correct solution, but I'm not expert in this area. Let's wait for other opinions